### DISCUSSION TITLE
how to use backgroundTokenizer

### DISCUSSION THREAD

**Discussion by ArunPatal** [ajaxorg/ace#4875] [Q&A]
Hello,

I want to observer few lines. Example => line 4, 5 & 6
```
some code
line4: <start>
line5: Some Code
line6: </end>
some code
```

Now the code i am using is

```
var tokenizer = ace.require("ace/tokenizer").Tokenizer;
var backgroundTokenizer = ace.require("ace/background_tokenizer").BackgroundTokenizer;
var background = new backgroundTokenizer(new tokenizer(), editor);
```
Now i am confused how to use it to observer tokens at line 4,5 & 6
Please provide a sample code showing how is it possible.

Thanks

**Comment by nightwing** [comment]
You can simply use `editor.session.getTokens(4)` and it will use the existing backgroundTokenizer to get the tokens on line 4

**Reply by ArunPatal** [reply]
Thanks,

I tested it. 

I stored it like
var line4 = editor.session.getTokens(4);
Later calling line4 show only those tokens which was stored at start.. 

Now what if row changes, how do i get new row of those tokens?

Also is there any event which tells if line got deleted or text changed?
